### PR TITLE
Feature/med 06

### PR DESCRIPTION
**Pull request by 8artosz**

### CODE REVIEW

File: src/main/java/com/github/bartosz/medicalclinic/controller/PatientController.java [hunk 1]
```diff
@@ -1,66 +1,62 @@
...
 import java.util.List;
 
 @RestController
 @RequestMapping("/patients")
 public class PatientController {
-    private final List<Patient> patients = new ArrayList<>();
+
+    private final PatientService patientService;
+
+    PatientController(PatientService patientService) {
```
Comment (konradgab) [inline src/main/java/com/github/bartosz/medicalclinic/controller/PatientController.java:16]: At our next meeting, we will talk about the integration between lombok and spring.

Comment (8artosz) [reply src/main/java/com/github/bartosz/medicalclinic/controller/PatientController.java:16]: Alright

File: src/main/java/com/github/bartosz/medicalclinic/service/PatientService.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+        if (password.equals(editedPatient.getPassword())) {
+            throw new PasswordAlreadyExistsException();
+        }
+
+        editedPatient.setPassword(password);
+    }
+
+    private Patient findPatientByEmail(String email) {
+        return patients.stream()
```
Comment (konradgab) [inline src/main/java/com/github/bartosz/medicalclinic/service/PatientService.java:76]: Before MED-06 ticket (MED-04) this solution was good but now to reuse this method it is better to change this implementation and return Optional<Patient> in order to throw specific exception in other service methods.

File: src/main/java/com/github/bartosz/medicalclinic/service/PatientService.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+                .findFirst()
+                .orElseThrow(PatientNotFoundException::new);
+    }
+
+    private boolean isPatientEditDataValid(Patient newPatient, String email) {
+        if (email.equals(newPatient.getEmail())) {
+            return true;
+        }
+
+        var newPatientWithGivenEmail = patients.stream()
```
Comment (konradgab) [inline src/main/java/com/github/bartosz/medicalclinic/service/PatientService.java:87]: for example here

**Pull request merged**